### PR TITLE
#58 support update and create datasets

### PR DESCRIPTION
**Pull request by marks**
please let me know how I should share the `SOCRATA_EMAIL` and `SOCRATA_PASSWORD` environment variables so the tests can run both locally and by travis-ci.org

Thank yo!

### CODE REVIEW

File: R/writeData.R [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#' 
+#' @param url - Socrata Open Data Application Program Interface (SODA) endpoint (JSON only for now)
+#' @param json_data_to_upload - JSON encoded data to update your SODA endpoint with
+#' @param http_verb - PUT or POST depending on update mode
+#' @param email - email associated with Socrata account (will need write access to dataset)
+#' @param password - password associated with Socrata account (will need write access to dataset)
+#' @param app_token - optional app_token associated with Socrata account
+#' @return httr a response object
+#' @importFrom httr GET
+#'
```
Comment (dmpe) [inline R/writeData.R]: #' importFrom httr GET authenticate add_headers
most probably then

Comment (dmpe) [inline R/writeData.R]: (for consideration) Idea for easier (nicer) debugging would be to add something like ... to the post function

File: R/writeData.R [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#' @param dataframe - dataframe to upload to Socrata
+#' @param dataset_json_endpoint - Socrata Open Data Application Program Interface (SODA) endpoint (JSON only for now)
+#' @param update_mode - "UPSERT" or "REPLACE"; consult http://dev.socrata.com/publishers/getting-started.html 
+#' @param email - email associated with Socrata account (will need write access to dataset)
+#' @param password - password associated with Socrata account (will need write access to dataset)
+#' @param app_token - optional app_token associated with Socrata account
+#'
+#' @return httr a response object
+#' @importFrom httr GET
+#'
```
Comment (dmpe) [inline R/writeData.R]: importFrom jsonlite toJSON

File: R/returnData.R [hunk 3]
```diff
@@ -101,7 +107,7 @@ getSodaTypes <- function(response) {
 #' @importFrom mime guess_type
 #' 
 #' @export
-read.socrata <- function(url, app_token = NULL) {
+read.socrata <- function(url, app_token = NULL, email = NULL, password = NULL) {
```
Comment (dmpe) [inline R/returnData.R]: are you sure this comes from the newest dev branch @tomschenkjr @marks  ? 
https://github.com/Chicago/RSocrata/blob/dev/R/returnData.R

Rebase maybe ?

**Comment by dmpe** [comment]
Great, thanks a lot. It looks fine to me. If this is working for you - with your password - then I am in favor of merging it (rebasing `dev` branch will be necessary though). 
Travis : just use skip on travis/cran/vol...if you want. 
Your changes can be even put in our next release which will happen anyway sometime in November, not October @tomschenkjr as previously planned.

**Comment by marks** [comment]
@dmpe you are right.. something didnt work right when I forked and created a branch off of `dev`... do you have any pointers on how to fix this? mucho gracias

**Comment by marks** [comment]
@dmpe I tried to follow https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request but ended up with quite a cluster on my hands. Do tests currently pass on `dev`? 

I am thinking I might just need to start over and make sure I actually begin from `dev`.

**Comment by dmpe** [comment]
I am not the right guy to ask about git :sob:  At worst, start again and use copy paste.

**Comment by marks** [comment]
Closing this in favor of re-doing my work on a branch that was actually based off `dev`: https://github.com/marks/rsocrata/tree/58-support-update-and-create-datasets-REDO

**Pull request closed without merging**